sktalk/corpus/conversation.py

```diff
@@ -68,3 +69,169 @@ def asdict(self):
+
+        Args:
+            index (int): The index of the utterance for which to provide context
+            before (int, optional): The time in ms preceding the utterance's begin. Defaults to 0.
+            after (int, optional): The time in ms following the utterance's end. Defaults to None,
+                which then assumes the same value as `before`.
+            exclude_utterance_overlap (bool, optional): If True, the duration of the
+                utterance itself is not used to identify overlapping utterances, and only
+                the window before or after the utterance is used. Defaults to False.
+
+        Returns:
+            Conversation: Conversation object without metadata, containing a reduced set of utterances
+        """
+        if index < 0 or index >= len(self._utterances):
+            raise IndexError("Index out of range")
+        if after is None:
+            after = before
+        try:
+            begin = self._utterances[index].time[0] - before
+            end = self._utterances[index].time[1] + after
+            if exclude_utterance_overlap and before == 0:  # only overlap with window following utterance
+                begin = self._utterances[index].time[1]
+            elif exclude_utterance_overlap and after == 0:  # only overlap with window preceding utterance
+                end = self._utterances[index].time[0]
+            returned_utterances = [
+                u for u in self._utterances if self.overlap(begin, end, u.time) or u == self._utterances[index]]
+        except (TypeError, IndexError):
+            returned_utterances = []
+        return Conversation(utterances=returned_utterances, suppress_warnings=True)
+
+    def count_participants(self, except_none: bool = False) -> int:
+        """Count the number of participants in a conversation
+
+        Importantly: if one of the utterances has no participant, it is counted
+        as a separate participant (None). If you want to exclude these, set
+        `except_none` to True.
+
+        Args:
+            except_none (bool, optional): if `True`, utterances without a participant are not counted. Defaults to `False`.
+
+        Returns:
+            int: number of participants
+        """
+        participants = [u.participant for u in self.utterances]
+        if except_none:
+            participants = [p for p in participants if p is not None]
+        return len(set(participants))
+
+    def _update(self, field: str, values: list, **kwargs):
+        """
+        Update the all utterances in the conversation with calculated values
+
+        This function also stores relevant arguments in the Conversation metadata.
+
+        Args:
+            field (str): field of the Utterance to update
+            values (list): list of values to update each utterance with
+            kwargs (dict): information about the calculation to store in the Conversation metadata
+        """
+        if len(values) != len(self.utterances):
```

Comment:
A minor thing, but I would suggest using the protected variable instead:

```suggestion
        if len(values) != len(self._utterances):
```